cpp/src/arrow/flight/CMakeLists.txt

```diff
@@ -291,7 +291,9 @@ if(ARROW_TESTING)
                 STATIC_INSTALL_INTERFACE_LIBS
                 ${ARROW_FLIGHT_TESTING_STATIC_INSTALL_INTERFACE_LIBS}
                 PRIVATE_INCLUDES
-                "${Protobuf_INCLUDE_DIRS}")
+                "${Protobuf_INCLUDE_DIRS}"
+                SHARED_PRIVATE_LINK_LIBS
+                GTest::gmock)
 
```

Comment:
This is the fix for the 12 build errors on the local environment. Rob and I have both seen this issue before. I am not sure why it is only visible in our local environments and not seen in CI